packages/docusaurus-mdx-loader/src/remark/admonitions/index.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import visit from 'unist-util-visit';
+import type {Transformer, Processor, Plugin} from 'unified';
+import type {Literal} from 'mdast';
+
+const NEWLINE = '\n';
+
+export type AdmonitionOptions = {
+  tag: string;
+  keywords: string[];
+};
+
+export const DefaultAdmonitionOptions: AdmonitionOptions = {
+  tag: ':::',
+  keywords: [
+    'secondary',
+    'info',
+    'success',
+    'danger',
+    'note',
+    'tip',
+    'warning',
+    'important',
+    'caution',
+  ],
+};
+
+function escapeRegExp(s: string): string {
+  return s.replace(/[-[\]{}()*+?.\\^$|/]/g, '\\$&');
+}
+
+function normalizeOptions(
+  options: Partial<AdmonitionOptions>,
+): AdmonitionOptions {
+  return {...DefaultAdmonitionOptions, ...options};
+}
+
+const plugin: Plugin = function plugin(
+  this: Processor,
+  optionsInput: Partial<AdmonitionOptions> = {},
+): Transformer {
+  const options = normalizeOptions(optionsInput);
+
+  const keywords = Object.values(options.keywords).map(escapeRegExp).join('|');
+  const tag = escapeRegExp(options.tag);
+  const regex = new RegExp(`${tag}(${keywords})(?: *(.*))?\n`);
+  const escapeTag = new RegExp(escapeRegExp(`\\${options.tag}`), 'g');
+
+  // the tokenizer is called on blocks to determine if there is an admonition
+  // present and create tags for it
+  function blockTokenizer(this: any, eat: any, value: string, silent: boolean) {
+    // stop if no match or match does not start at beginning of line
+    const match = regex.exec(value);
+    if (!match || match.index !== 0) {
+      return false;
+    }
+    // if silent return the match
+    if (silent) {
+      return true;
+    }
+
+    const now = eat.now();
+    const [opening, keyword, title] = match;
+    const food = [];
+    const content = [];
+
+    let newValue = value;
+    // consume lines until a closing tag
+    let idx = newValue.indexOf(NEWLINE);
```

Comment:
Not sure if this can be done in this PR or in a followup. But i would suggest to support nested Admonitions aswell, s.t. something like

```md
:::info Weather
On nice days, you can enjoy swimming in the lake.

:::danger Storms
Take care of summerstorms...
:::

:::
```

would work as expected.

To do this, we would need to keep track of the nesting level. I already did something similar for https://github.com/elviswolcott/remark-admonitions/issues/41 here: https://github.com/elviswolcott/remark-admonitions/pull/42/files#diff-92bbac9a308cd5fcf9db165841f2d90ce981baddcb2b1e26cfff170929af3bd1R170

Would be nice to see this supported here aswell.

Reply:
Thanks for that info @lebalz Let's do that as a follow-up because this one is supposed to be a simple port.